src/NServiceBus.Core.Tests/StructConventionsTests.cs

```diff
@@ -113,17 +137,24 @@ static void InspectWhetherStructContainsWritableProperties(Type type, List<strin
             }
 
             if (mutabilityRules.Count > 1)
+            {
                 violatedRules.AddRange(mutabilityRules);
+
+                return true;
+            }
+
+            return false;
         }
 
         static void InspectSizeOfStruct(Type type, List<string> violatedRules)
         {
             try
             {
-                var sizeOf = Marshal.SizeOf(type);
-                if (IsLargerThanSixteenBytes(sizeOf))
+                var size = Marshal.SizeOf(type);
```

Comment:
can you  share some background on the Marshal.SizeOf differences across the platforms?